examples/guides/services/healthcare-service/tests/clemency-hospital-service-test.bal

```diff
@@ -0,0 +1,547 @@
+#
+# + dataset - dataset Parameter Description
+@test:Config 
+{
+    dataProvider: "testReserveAppointmentDataProvider",
+     dependsOn: ["testAddDoctor"]
+}
+
+function testReserveAppointment(json dataset, json expectedStrings) 
+{
+    // set the json payload
+    http:Request request = new;
+    json payload = dataset;
+    request.setPayload(payload);
+
+    // sending the post request to the endpoint
+    var response = clientEPclemency->post("/surgery/reserve", request);
+
+    if(response is http:Response)
+    {
+        string doctor= dataset.doctor.toString();
+        var responseStatusCode = response.statusCode;
+        var expectedStatusCode = expectedStrings.statusCode;
+        var expectedResponseText = expectedStrings.responseMessage;
+
+        if(doctor == "T Uyanage")
+        {
+            var responsePayload = response.getTextPayload();
+            test:assertEquals(responsePayload, expectedResponseText, msg = "Assertion Failed for Doctor " +doctor);
+            test:assertEquals(responseStatusCode, expectedStatusCode, msg = "Status code mismatch!");
+        }
+        else
+        {
+            var resonsePayload = response.getJsonPayload();
+            var expectedIncludesAppoinmentNumber = expectedStrings.appoinmentNumber;
+            var expectedIncludesDoctorAvailibility = expectedStrings.doctorAvailibility;
+            var expectedIncludesDoctorFee = expectedStrings.doctorFee;
+
+            if(resonsePayload is json)
+            {
+                boolean responseIncludesAppoinmentNumber = false;
+                boolean responseIncludesDoctorAvailibility = false;
+                boolean responseIncludesDoctorFee = false;
+
+                // Verifying if response json payload includes appointmentNumber, availability and fee
+                if(resonsePayload.toString().contains("appointmentNumber"))
+                {
+                    responseIncludesAppoinmentNumber = true;
+                }
+                if(resonsePayload.doctor.toString().contains("availability"))
+                {
+                    responseIncludesDoctorAvailibility = true;
+                }
+                if(resonsePayload.toString().contains("fee"))
+                {
+                    responseIncludesDoctorFee = true;
+                }
+                test:assertEquals(response.statusCode, expectedStatusCode, msg = "Status Code mismatch!");
+                test:assertEquals(responseIncludesAppoinmentNumber, expectedIncludesAppoinmentNumber, msg = "Appoinment number is not as expected");
+                test:assertEquals(responseIncludesDoctorAvailibility, expectedIncludesDoctorAvailibility, msg = "Doctor availability is not as expected");
+                test:assertEquals(responseIncludesDoctorFee, expectedIncludesDoctorFee, msg = "Doctor fee is not as expected");  
+            }
+            else
+            {
+                test:assertFail(msg = "Test Failed!");
```

Comment:
Shall we have a proper failure message like 'invalid payload'?

Reply:
Done.